the-collab-lab/tcl-27-smart-shopping-list
# Dp sk add firebase

Co-Authored-by: Dani Poplin

## Description
What we did: 
- imported firebase into our project and enabled the client to send a receive from the database.

Why we did it:
- to allow user to persist data and submit an item to their grocery list.

How we did it:
- separated our code into individual components to promote readability and to follow the single source of truth convention. 
- Added 2 new components
    - GroceryContainer.js
    - GroceryCard.js

## Acceptance Criteria
* The following have been added as project dependencies: firebase & react-firebase-hooks
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database
[Add Firebase](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/1)

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria
* From your terminal, pull down this branch with git pull origin dp-sk-add-firebase and check that branch out with git checkout dp-sk-add-firebase
* Then yarn or npm to install the newly added dependencies locally and yarn or npm start to launch the app.
* Click the 'submit an item' button. Make sure it adds it to the list and make sure that item appears in the items collection of the database [here](https://console.firebase.google.com/project/tcl-27-shopping-list/firestore/data/~2Fgroceries).
